src/NetAnalyzers/Core/Microsoft.NetCore.Analyzers/Runtime/PreferStreamAsyncMemoryOverloads.Fixer.cs

```diff
@@ -145,7 +150,11 @@ public sealed override async Task RegisterCodeFixesAsync(CodeFixContext context)
             }
             SyntaxNode newInvocationExpression = generator.InvocationExpression(asyncMethodNode, nodeArguments).WithTriviaFrom(streamInstanceNode);
 
-            SyntaxNode newRoot = generator.ReplaceNode(root, invocation.Syntax, newInvocationExpression.WithTriviaFrom(invocation.Syntax));
+            bool containsSystemImport = IsSystemNamespaceImported(generator.GetNamespaceImports(root));
+
+            SyntaxNode rootWithImports = containsSystemImport ? root : generator.AddNamespaceImports(root, generator.NamespaceImportDeclaration("System").WithAddImportsAnnotation());
+            SyntaxNode newRoot = generator.ReplaceNode(rootWithImports, invocation.Syntax, newInvocationExpression.WithTriviaFrom(invocation.Syntax));
```

Comment:
If I put a breakpoint here, the `newRoot` contains the `using System;` line added at the end of the list of imports, as expected, yet the unit test fail with the error I shared in my other comment.